test/integ/pandas/test_arrow_pandas.py

```diff
@@ -591,99 +705,54 @@ def validate_pandas(
-                            "new value is {}, values are not equal".format(i, j, c_new)
-                        )
-                    else:
-                        if data_type == "float":
-                            c_case = float(cases[i])
-                        elif data_type == "decimal":
-                            c_case = Decimal(cases[i])
-                        elif data_type == "date":
-                            c_case = datetime.strptime(cases[i], "%Y-%m-%d").date()
-                        elif data_type == "time":
-                            time_str_len = 8 if scale == 0 else 9 + scale
-                            c_case = cases[i].strip()[:time_str_len]
-                            c_new = str(c_new).strip()[:time_str_len]
-                            assert c_new == c_case, (
-                                "{} row, {} column: original value is {}, "
-                                "new value is {}, "
-                                "values are not equal".format(i, j, cases[i], c_new)
-                            )
-                            break
-                        elif data_type.startswith("timestamp"):
-                            time_str_len = 19 if scale == 0 else 20 + scale
-                            if timezone:
-                                c_case = pandas.Timestamp(
-                                    cases[i][:time_str_len], tz=timezone
-                                )
-                                if data_type == "timestamptz":
-                                    c_case = c_case.tz_convert("UTC")
-                                    c_case = c_case.tz_localize(None)
-                            else:
-                                c_case = pandas.Timestamp(cases[i][:time_str_len])
-                            assert c_case == c_new, (
-                                "{} row, {} column: original value is {}, new value is {}, "
-                                "values are not equal".format(i, j, cases[i], c_new)
-                            )
-                            break
-                        else:
-                            c_case = cases[i]
-                        if epsilon is None:
-                            assert c_case == c_new, (
-                                "{} row, {} column: original value is {}, new value is {}, "
-                                "values are not equal".format(i, j, cases[i], c_new)
-                            )
-                        else:
-                            assert abs(c_case - c_new) < epsilon, (
-                                "{} row, {} column: original value is {}, "
-                                "new value is {}, epsilon is {} \
-                            values are not equal".format(
-                                    i, j, cases[i], c_new, epsilon
-                                )
-                            )
+def validate_pandas_with_telemetry(
```

Comment:
This function doesn't really do anything either.
What if we had the `capture_sf_telemetry` in the test and added an assert in the test instead of passing things down to helper functions instead

Reply:
The reason to use a do_validate_pandas is, when we call to it, the connection and telemetry objects are ready. Before that, there is no good way to get these two object out of fixture context manager. So keep the code to load fixture outside in validate_pandas_with_telemetry and validate_pandas, and have the actual meat in do_validate_pandas I think is the clean way to use fixture.

Reply:
You could instead use the fixtures in the test itself and pass the created objects down into the helper functions instead of passing the fixtures down.

I think it'd be helpful for you to trace out what `conn_cnx` really does here https://github.com/snowflakedb/snowflake-connector-python/blob/641204933c4f7e125f3af064b67354199943d61c/test/integ/conftest.py#L221

The fixtures have nothing to do with anything once you're in a test, they pass in whatever the `fixture` function returns.